functions/callPythonFunction/pythonFunctions/create_certificates.py

```
@@ -295,19 +419,36 @@ def prepare_text_content(self, enrollment, image):
+                        "full_name": f"{enrollment['User']['firstName'].upper()} {enrollment['User']['lastName'].upper()}",
+                        "course_name": enrollment["Course"]["title"],
+                        "semester": enrollment["Course"]["Program"]["title"],
+                        "template": image,
+                        "ECTS": enrollment["Course"]["ects"],
+                        "program_title": enrollment["Course"]["Program"]["title"],
+                        "successful_participations": successful_participations
+                    }
+                else:
+                    # Regular achievement certificate
+                    if not enrollment.get('Course') or not enrollment.get('Course', {}).get('learningGoals'):
+                        raise CertificateError("Missing required course or learning goals data", "MISSING_COURSE_DATA")
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)